behat-tests/features/acquia_lift_element_variations.feature

```diff
@@ -1,170 +1,198 @@
 Feature: Personalize elements variations can be edited for an existing campaign.
   In order to manage element variations in context
   As a site marketer
   I want the ability to add, edit, and delete existing personalize element variations in context.
 
-  @api @javascript @campaign
   Scenario: Add personalize elements to a campaign
+    # I have a campaign.
+    # I login with the marketer role.
+    # I am on an article page.
     Given "acquia_lift" agents:
-      | machine_name                    | label |
+      | machine_name                    | label                           |
       | testing-campaign-add-variations | Testing campaign add variations |
-    And I am logged in as a user with the "Marketer" role
+    And I am logged in as a user with the "access administration pages,access toolbar,administer visitor actions,manage personalized content" permission
     And I am on the homepage
+    And I am viewing an "Article" content:
+      | title | Test Article Title - Original |
     When I click "Acquia Lift" in the "menu" region
     And I wait for AJAX to finish
     Then I should see the link "Campaigns" in the "lift_tray" region
+
+    # I open the variation set's menu.
     When I hover over "Campaigns" in the "lift_tray" region
     And I click "Testing campaign add variations" in the "lift_tray" region
-    Then I should see the link "Variation Sets" visible in the "lift_tray" region
+    Then I should visibly see the link "Variation Sets" in the "lift_tray" region
     And I should see "0" for the "variation set" count
+
+    # I bring up the "Add variation set" interface.
     When I hover over "Variation Sets" in the "lift_tray" region
     Then I should see the link "Add variation set" in the "lift_tray" region
     Then I should see the link "All variation sets" in the "lift_tray" region
     When I click "Add variation set"
-    And I wait for AJAX to finish
     Then I should see the modal with title "Add a variation set"
     And I should see the link "Webpage elements" in the "modal_content" region
     And I should see the link "Drupal blocks" in the "modal_content" region
     When I click "Webpage elements" in the "modal_content" region
-    And I wait for AJAX to finish
     Then I should not see the modal
-    When I click "#site-name a span" element in the "page_content" region
-    And I wait for AJAX to finish
-    Then I should see the text "<SPAN>" in the "dialog_variation_type" region
+
+    # I add a new variation set.
+    When I click "#page-title" element in the "page_content" region
+    Then I should see the text "<H1>" in the "dialog_variation_type" region
     When I click "Edit text" in the "dialog_variation_type" region
-    And I wait for AJAX to finish
     Then I should not see the variation type dialog
-    And I should see the text "Edit text: <SPAN>" in the "dialog_variation_type_form" region
-    And the "personalize_elements_content" field should contain the site title
```

Comment:
Yes I have had the same reasoning as yours, that's why this one has been so hesitant for me. The only reason I noted it is because HTML ignores the leading and trailing space chars anyway, and those space chars are frequently added for DOM indentation, so I thought trimming them was actually bettering user experience.

On the other hand, I could not figure out a contain, the best solution I have so far is to introducing a new step: 
```Gherkin
@Then 
And the "personalize_elements_content" field should contain the same text as "#page-title" element
```